plugins/commands.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+const joi = require('joi');
+const boom = require('boom');
+const requestApi = require('request');
+const SCHEMA_COMMAND_NAMESPACE = joi.string().label('Command namespace');
+const SCHEMA_COMMAND_NAME = joi.string().label('Command name');
+const SCHEMA_COMMAND_VERSION = joi.string().label('Command version');
+const DEFAULT_TTL = 21 * 24 * 60 * 60 * 1000; // 3 weeks
+const DEFAULT_BYTES = 1024 * 1024 * 1024; // 1GB
+
+/**
+ * Determine if a pipeline can publish the command
+ * @method canPublishCommand
+ * @param  {String}  namespace  Command namespace
+ * @param  {String}  name       Command name
+ * @param  {Integer} pipelineId pipelineId
+ * @param  {String}  baseurl    API URL
+ * @return {Promise}
+ */
+function canPublishCommand(namespace, name, pipelineId, baseurl) {
+    return new Promise((resolve, reject) => {
+        requestApi({
+            uri: `${baseurl}/v4/commands`,
+            method: 'GET',
+            json: true
+        },
+        (err, response) => {
+            if (err) {
+                return reject(err);
+            }
+
+            return resolve(response);
+        });
+    }).then((response) => {
+        const commands = response.body.filter(element =>
+            element.namespace === namespace && element.name === name
+        );
+
+        if (commands.length === 0) {
+            return Promise.resolve(true);
```

Comment:
Does this mean sever can publish command even if there is no command?

Reply:
It means that there are no `namespace/name` commands. In other words, it is first time to publish `namespace/name` command. Therefore, `namespace/name` command is allowed to be published.

Reply:
Understood. Thank you for explanation.

Reply:
I'd like you to add comment about this context because it might be difficult to analogize it only from program.

Reply:
Added comment.